joomla/joomla-cms
# [4.1] Implement access levels for media manager folder

### Summary of Changes
This PR allows to define access levels for the different folder entries, adds a level to the folder selection and fixes a bug with subfolders in the local folder management in the media manager.

![grafik](https://user-images.githubusercontent.com/1229869/139416861-3b318da6-28a7-4e9c-a0a4-44118be7c37b.png)

![grafik](https://user-images.githubusercontent.com/1229869/139417588-a5d4ec36-63bb-451d-abe6-dbf553022a69.png)

(2nd folder not visible, because user is logged in)

### Testing Instructions
1. Apply PR
2. Go to the "FileSystem - Local" plugin
3. Set some folders + permissions
4. Create users to check the permissions (by e.g. opening the media select for intro images in an article)


### Actual result BEFORE applying this Pull Request
No access level option available


### Expected result AFTER applying this Pull Request
Only folders with the correct access level are shown
